classes/Kohana/Jam/Field/Integer.php

```diff
@@ -32,5 +32,36 @@ public function set(Jam_Validated $model, $value, $is_changed)
 
 		return $value;
 	}
+	
+	/**
+	 * Like Jam_Field::_default, but it won't convert 0 and 0.0 to empty_value
+	 *
+	 * @param  Jam_Validated $model
+	 * @param  mixed $value
+	 * @return array 1st element the converted value; 2nd element a boolean
+	 * indicating if the value should not be processed further
+	 */
+	protected function _default(Jam_Validated $model, $value)
+	{
+		$return = FALSE;
+
+		$value = $this->run_filters($model, $value);
+
+		// Convert empty values to NULL, if needed
+		if ($this->convert_empty AND empty($value) AND $value !== 0 AND $value !== 0.0 AND $value !== "0" AND $value !== "0.0")
```

Comment:
You'd want to use a filter on the field. For example you can use the `intval` function as a filter to convert the string values to integer.

On your second question - it depends on what the field represents. If it's a counter - not allowing NULL is correct and it should be represented in the database schema. If it's a boolean - the same. If it's a price - it depends on your business logic. 0 means free, NULL means not set and you'd set those depending on your business requirements.

You could also try using `Jam_Field_Decimal`.